### DISCUSSION TITLE
Enhanced Go to Matching Bracket command

### DISCUSSION THREAD

**Discussion by pyscripter** [pyscripter/SynEdit#46] [Ideas]
Currently the ecMatchBracket command works like this:
*star pairs show matching brackets*

```
*[(some text)*]
[*(some text*)]
[(*some text)] no matching bracket
[(some text)]* no matching bracket
```

In Scintilla (Notepad+ and many other editors) and PyScripter it works like this:
```
*[(some text)]*
[*(some text)*]
[(*some text*)] 
```
Notice that if you start outside a bracket you stay outside the matching bracket and if you start inside a bracket you stay inside the matching bracket.

We could also add an ecSelMatchBracket command.

@MShark67,@vincentparrett,@JaFi-cz  What do you think?

**Comment by vincentparrett** [comment]
It's not a massive change in behaviour so wouldn't have any objections to it.

**Comment by JaFi-cz** [comment]
I am for behavior to match it as second case.

**Comment by MShark67** [comment]
Second case looks good to me!

**Comment by pyscripter** [comment]
Implemented.
ecSelMatchBracket command added.

Shortcuts:
ecMatchBracket:  Ctrl + ']'
ecSelMatchBracket: Shft + Ctrl + ']'
